BlackMarket/app/SignUp/SignUpCardView.swift

```diff
@@ -0,0 +1,211 @@
+//
+//  SignUpCardView.swift
+//  BlackMarket
+//
+//  Created by Tarek Radovan on 06/12/2022.
+//
+
+import SwiftUI
+import ComposableArchitecture
+
+struct SignUpCardDomain: ReducerProtocol {
+  
+  struct State: Equatable {
+    static func == (lhs: SignUpCardDomain.State, rhs: SignUpCardDomain.State) -> Bool {
+      return true
+    }
+   
+    var nameState = BMTextFieldDomain.State(
+      validations: [.nonEmpty],
+      placeholder: LocalizedString.SignUpCardView.nameTextfieldPlaceholder,
+      errorMessage: LocalizedString.SignUpCardView.nameTextfieldError,
+      title: LocalizedString.SignUpCardView.nameTextfieldTitle
+    )
+    var emailState = BMTextFieldDomain.State(
+      validations: [.email, .nonEmpty],
+      placeholder: LocalizedString.SignUpCardView.emailTextfieldPlaceholder,
+      errorMessage: LocalizedString.SignUpCardView.emailTextfieldError,
+      title: LocalizedString.SignUpCardView.emailTextfieldTitle
+    )
+    var passwordState = BMTextFieldDomain.State(
+      validations: [
+        .custom(isValid: { value in
+          value.count >= Constants.Value.minimumPasswordLength
+        })
+      ],
+      placeholder: LocalizedString.SignUpCardView.passwordTextfieldPlaceholder,
+      errorMessage: LocalizedString.SignUpCardView.passwordTextfieldError,
+      title: LocalizedString.SignUpCardView.passwordTextfieldTitle,
+      isSecure: true
+    )
+    
+    var dataPolicyLink: AttributedString {
+      LocalizedString.SignUpCardView.dataPolicy.transformToLink(
+        withURL: Constants.Path.dataPolicy
+      ) ?? ""
+    }
+    var cookiesPolicyLink: AttributedString {
+      LocalizedString.SignUpCardView.cookiesPolicy.transformToLink(
+        withURL: Constants.Path.cookiesPolicy
+      ) ?? ""
+    }
```

Comment:
instead of computed vars, it seems that these can be initialized once only.

Reply:
Fixed in [this PR](https://github.com/tarruk/blackMarket/pull/7)